src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/If_AllowsSideEffects.txt

```diff
@@ -0,0 +1,13 @@
+﻿#SETUP: EnableExpressionChaining
+
+>> If(false, "abc1", {x:3})
+{x:3}
```

Comment:
Can we add a comment here to explain why this is ok?
(possibly reference IfFunction.IsArgTypeInconsequential) 

"This If() expression is allowed to have mismatching types between arg1 and arg2 because a) it's behavior, b) it's verified nobody is actually using the return value."

Reply:
Done.